hack/verify-file-sizes.sh

```diff
@@ -24,6 +24,22 @@ cd "${KUBE_ROOT}"
 # Files larger than 1MB need to be allowed explicitly.
 maxsize=$((1 * 1024 * 1024))
 
+run_stat() {
+    local file="$1"
+    case "$(uname -s)" in
+        Darwin)
+            stat -f %z "$file" | tr -d '\n'
```

Comment:
If this line works well on Linux, why not use it directly in Line 61?

Reply:
@carlory Thank you for your review.
Regarding the options for the stat command, I tried it on Linux, and here's what I found:

```
$ touch samplefile

$ stat -f %z samplefile | tr -d '\n'
stat: cannot read file system information for '%z': No such file or directory
  File: "samplefile"    ID: 0        Namelen: 255     Type: fuseblkBlock size: 1048576    Fundamental block size: 4096Blocks: Total: 59840624   Free: 18538697   Available: 18538697Inodes: Total: 744080125  Free: 741547880

$ stat --printf %z ./samplefile | tr -d '\n'
2024-04-03 03:25:17.374582976 +0000

$ stat --printf %s ./samplefile
0
```

As shown above, an error occurred with `stat -f %z`, so I am switching options and such depending on whether the environment is Linux or Mac OS X.

Reply:
this sort of thing tends to cause problems because even on darwin you may have gnu coreutils installed, check e.g. when we require gnu-sed.

`wc -c < "$file"` would work portably

Reply:
@BenTheElder Thank you for your comment.
I have changed the implementation to use `wc -c < "$file"`. Please review it when you have time.

Just to be safe, I've attached the test results below after verifying the operation on both Mac OS X and Linux:

**Mac OS X**

```
echo "abc" > samplefile
size=$(wc -c < samplefile)
echo "size: $size"
for i in 3 4
do
  maxsize=$i
  if [ "${size}" -gt "$maxsize" ]; then
    echo "size is greater than maxsize"
  else
    echo "size is less than or equal to maxsize"
  fi
done
size:        4
size is greater than maxsize
size is less than or equal to maxsize
```

**Linux**

```
echo "abc" > samplefile
size=$(wc -c < samplefile)
echo "size: $size"
for i in 3 4
do
  maxsize=$i
  if [ "${size}" -gt "$maxsize" ]; then
    echo "size is greater than maxsize"
  else
    echo "size is less than or equal to maxsize"
  fi
done
size: 4
size is greater than maxsize
size is less than or equal to maxsize
```

Reply:
also confirming:

this test took
this pr: 3s: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/124156/pull-kubernetes-verify/1781407536046411776
random PR without this change: 6s: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/119589/pull-kubernetes-verify/1781378276028059648

so no performance issues